### PR TITLE
Add error when passing empty array to `c_ptrTo`

### PR DESCRIPTION
**Pull request by bmcdonald3**
When `c_ptrTo` is passed an empty array, today we will get a halt for OOB, but that error message can be confusing, so this PR adds an explicit check for that in `c_ptrTo` that provides a more helpful error message.

### CODE REVIEW

File: modules/standard/CTypes.chpl [hunk 1]
```diff
@@ -496,6 +496,11 @@ module CTypes {
     if (arr._value.locale != here) then
       halt("c_ptrTo() can only be applied to an array from the locale on which it lives (array is on locale " + arr._value.locale.id:string + ", call was made on locale " + here.id:string + ")");
 
+    if boundsChecking {
+      if (arr.size == 0) then
+        halt("Can’t take a C pointer to an array with 0 elements.");
```
Comment (bradcray) [inline modules/standard/CTypes.chpl:501]: Thanks for jumping on this, Ben!  I'd be fine with this going in as-is.  If I were to wordsmith slightly, I might say "Can't create a C pointer for an array with 0 elements."

Comment (bmcdonald3) [reply modules/standard/CTypes.chpl]: Sounds good to me; updated.

**Comment by bradcray** [comment]
@bmcdonald3 : I haven't done anything to review this deeply other than glance through the code.  It looks like a reasonable no-brainer to me.  Given that she's working on c_ptrTo() in other contexts and is more deliberate than I am, it might make sense to have @riftEmber give this a look as well to make sure I haven't missed anything dumb.

**Review by riftEmber — APPROVED** [review]
LGTM!

**Pull request merged**